### PR TITLE
MINOR: [Dev] Add kevingurney to CODEOWNERS for `/matlab/`

### PR DESCRIPTION
**Pull request by kevingurney**
Follow up to https://github.com/apache/arrow/issues/36474#issuecomment-1623128986.

Marking this as MINOR following https://github.com/apache/arrow/pull/36504 and https://github.com/apache/arrow/pull/36468.

### CODE REVIEW

File: .github/CODEOWNERS
Comment (assignUser) [inline .github/CODEOWNERS:45]:
```suggestion
/matlab/ @kevingurney
```
As I only added myself to catch any stray matlab PRs that need you tagged on I can now be removed, congrats again!

Comment (kevingurney) [reply .github/CODEOWNERS:45]: Thank you, @assignUser! :)

**Review by raulcd — APPROVED** [review]
@kevingurney do you want to merge?

**Comment by kevingurney** [comment]
@raulcd - sure! I'll merge this in. Thanks!

**Comment by kevingurney** [comment]
@raulcd - I noticed that after merging in this PR using the GitHub UI, that the commit message doesn't have "Authored-by" or "Signed-off-by" like other commits do. Is this something that should be added manually to the commit message when squashing?

I looked through the development documentation for information on this topic, but couldn't seem to find anything. I recall that there are some automated scripts that get run during PR merging that add things like "Co-authored-by" when there are commits from multiple authors associated with a PR, but I'm not sure about "Signed-off-by".

I apologize for my lack of knowledge on this topic.

**Comment by assignUser** [comment]
Ah we don't use the ui to merge (for a number of reasons of technical/historic reasons), the merge script creates the standardized commit message etc. see https://github.com/apache/arrow/blob/41e08795864b96f8fbbc9c42975d82e3bb0bb6d1/dev/README.md#how-to-merge-a-pull-request

Shouldn't be an issue with minor PR so don't worry!

**Comment by kevingurney** [comment]
Ah, I am very sorry about the oversight on my part! I'm not sure how I missed that documentation. I'll be sure to use the merge script in the future. Thank you for helping clear this up for me!

My sincere apologies again for merging this PR incorrectly!

**Comment by kevingurney** [comment]
I did a quick search online about ways to disable the green merge button in the UI to prevent others from accidentally making the same mistake in the future. Unfortunately, it seems like there isn't an easy way to do this.

There do appear to be a few "workarounds" suggested online (e.g. [this article](https://www.jessesquires.com/blog/2021/08/24/useful-label-based-github-actions-workflows/#do-not-merge)), such as creating a CI workflow that will unconditionally fail if a certain label is present on the PR. However, this has the major downside that the default PR CI status would always include a failing workflow, which does not feel ideal.

Perhaps, a more straightforward solution would be to create some kind of "Getting Started as a Committer" guide (or something to that effect), with https://github.com/apache/arrow/blob/41e08795864b96f8fbbc9c42975d82e3bb0bb6d1/dev/README.md#how-to-merge-a-pull-request included as "required reading".

If that sounds helpful, I could follow up with a PR for that and work with other community members to ensure that it contains any other important references and is included in any materials that are shared with new committers.

**Comment by kevingurney** [comment]
For reference, I captured the idea of creating a "New Committer Guide" in #36644.

**Comment by westonpace** [comment]
That's a great idea.  If it's any consolation [I did an accidental push-to-main within a month of getting write privileges](https://lists.apache.org/list?dev@arrow.apache.org:lte=3y:force%20push%20accidental%20commit) :laughing:

**Comment by assignUser** [comment]
Hm as we now use the api to merge we can probably use the protected branches feature mentioned in the thread? Thanks for sharing :D

**Pull request merged**
